pkgs/servers/uwsgi/default.nix

```diff
@@ -0,0 +1,76 @@
+{ stdenv, lib, fetchurl, pkgconfig, jansson
+, plugins
+, pam, withPAM ? stdenv.isLinux
+, systemd, withSystemd ? stdenv.isLinux
+, python2, python3, ncurses
+}:
+
+let pythonPlugin = pkg : { name = "python${if pkg ? isPy2 then "2" else "3"}";
+                           interpreter = pkg;
+                           path = "plugins/python";
+                           deps = [ pkg ncurses ];
+                           install = ''
+                             install -Dm644 uwsgidecorators.py $out/${pkg.sitePackages}/uwsgidecorators.py
+                             ${pkg.executable} -m compileall $out/${pkg.sitePackages}/
+                             ${pkg.executable} -O -m compileall $out/${pkg.sitePackages}/
+                           '';
+                         };
+    available = [ (pythonPlugin python2)
+                  (pythonPlugin python3)
+                ];
+     needed = builtins.filter (x: lib.any (y: x.name == y) plugins) available;
```

Comment:
This is codestyle oftenly used in Haskell, sorry for this. I'll fix this to stick to Nix standards.